python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py

```diff
@@ -801,35 +805,145 @@ def airbyte_cloud_resource(context) -> AirbyteCloudResource:
     _access_token_timestamp: Optional[float] = PrivateAttr(default=None)
 
-    def __init__(
-        self,
-        workspace_id: str,
-        client_id: str,
-        client_secret: str,
-    ):
-        self.workspace_id = workspace_id
-        self.client_id = client_id
-        self.client_secret = client_secret
-
     @property
     @cached_method
     def _log(self) -> logging.Logger:
         return get_dagster_logger()
 
     @property
     def api_base_url(self) -> str:
-        raise NotImplementedError()
+        return f"{AIRBYTE_API_BASE}/{AIRBYTE_API_VERSION}"
+
+    @property
+    def all_additional_request_params(self) -> Mapping[str, Any]:
+        return deep_merge_dicts(self.authorization_request_params, self.user_agent_request_params)
+
+    @property
+    def authorization_request_params(self) -> Mapping[str, Any]:
+        # Make sure the access token is refreshed before using it when calling the API.
+        if self._needs_refreshed_access_token():
+            self._refresh_access_token()
+        return {
+            "headers": {
+                "Authorization": f"Bearer {self._access_token_value}",
+            }
+        }
+
+    @property
+    def user_agent_request_params(self) -> Mapping[str, Any]:
+        return {
+            "headers": {
+                "User-Agent": "dagster",
+            }
+        }
+
+    def _refresh_access_token(self) -> None:
+        response = check.not_none(
+            self._make_request(
+                method="POST",
+                endpoint="/applications/token",
+                base_url=self.api_base_url,
+                data={
+                    "client_id": self.client_id,
+                    "client_secret": self.client_secret,
+                },
+                # Must not pass the bearer access token when refreshing it.
+                include_additional_request_params=False,
+            )
+        )
+        self._access_token_value = str(response["access_token"])
+        self._access_token_timestamp = datetime.now().timestamp()
+
+    def _needs_refreshed_access_token(self) -> bool:
+        return (
+            not self._access_token_value
+            or not self._access_token_timestamp
+            or self._access_token_timestamp
+            <= (
+                datetime.now() - timedelta(seconds=AIRBYTE_CLOUD_REFRESH_TIMEDELTA_SECONDS)
+            ).timestamp()
+        )
 
     def _make_request(
-        self, method: str, endpoint: str, data: Optional[str] = None
+        self,
+        method: str,
+        endpoint: str,
+        base_url: str,
+        data: Optional[Mapping[str, Any]] = None,
+        include_additional_request_params: bool = True,
     ) -> Mapping[str, Any]:
-        raise NotImplementedError()
+        """Creates and sends a request to the desired Airbyte REST API endpoint.
+
+        Args:
+            method (str): The http method to use for this request (e.g. "POST", "GET", "PATCH").
```

Comment:
nit: can probably use the requests enum here?

Reply:
The mock response library provides an enum, but the request library doesn't - the request library [still uses strings](https://github.com/psf/requests/blob/v2.32.3/src/requests/api.py) in the implementation.